### PR TITLE
Add util method to convert image url to path and vice versa.

### PR DESCRIPTION
**Pull request by huaiyan**
# Description

Add new util class to convert image url to path and vice versa.
Runtime will need it for universal link in async flow test.
When use this util class, invoker needs to pass in yaml file path, which could make sure contract version is a concept invisible for external.

Introduce `AbstractMultimediaFormatConverter` for contract specific logic.
Expect to inherit it for each future contract.
Please help about thinking a better name than `MultimediaFormatConverter20231201`

New util method addition:
* [`src/promptflow/promptflow/_utils/multimedia_format_converter.py`](diffhunk://#diff-b0656d341f321e5e03d58a69d01e23e98c0f51fa5503a8b99471babc77e70b91R1-R130): Added the `multimedia_format_converter.py` module with classes for converting multimedia expressions between path and URL formats.

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow/promptflow/_utils/multimedia_utils.py [hunk 1]
```diff
@@ -224,12 +222,50 @@ def load_multimedia_data(inputs: Dict[str, FlowInputDefinition], line_inputs: di
...
+        for key in image_dict:
+            format, resource = _get_multimedia_info(key)
+            if resource == "path":
+                path = image_dict[key]
+                url = converter(path)
+                return {f"data:image/{format};url": url}
+    return image_dict
+
+
+def convert_multimedia_url_to_path_recursively(value: Any, convert_url_to_path: Callable[[str], str]):
```
Comment (y1811) [inline src/promptflow/promptflow/_utils/multimedia_utils.py:250]: A question: why not also define this convert_url_to_path in utils?

Comment (huaiyan) [reply src/promptflow/promptflow/_utils/multimedia_utils.py:250]: That should be a real client sdk implementation. Such as using Compute Instance Identity to download url.
So, couldn't define in there.

**Comment by huaiyan** [comment]
Will add more complex structure, to guarantee runtime don't need to care about contract upgrade.

File: src/promptflow/promptflow/_utils/multimedia_format_converter.py [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+        return self._create_multimedia_data_from_path_or_url(origin_data, path, self.PATH_RESOURCE_TYPE)
+
+    def create_multimedia_data_from_url(self, origin_data: Any, url: str):
+        return self._create_multimedia_data_from_path_or_url(origin_data, url, self.URL_RESOURCE_TYPE)
+
+
+class MultimediaConverter:
+    def __init__(self, flow_file: Path):
+        """
+        Initialize the MultimediaConverter.
```
Comment (thy09) [inline src/promptflow/promptflow/_utils/multimedia_format_converter.py:66]: We may add more comments why we need this.

**Review by thy09 — APPROVED** [review]

File: src/promptflow/promptflow/_utils/multimedia_format_converter.py [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+
+# Format converter to switch multimedia expression between path and url for each contract version.
+class AbstractMultimediaFormatConverter:
+    # Check if the original_data is a multimedia format according to current contract version.
+    def is_multimedia_format(self, original_data: Any):
+        raise NotImplementedError()
+
+    # Get the path from multimedia data. Below is example for 20231201 version:
+    # {"data:image/jpg;path": "logo.jpg"} -> "logo.jpg"
+    def get_path_from_multimedia_data(self, original_data: Any):
```
Comment (thy09) [inline src/promptflow/promptflow/_utils/multimedia_format_converter.py:16]: maybe `get_path_from_dict`?

**Pull request merged**